owasp-top10-2017-apps/a6/stegonography/README.md

```diff
@@ -0,0 +1,113 @@
+# Stegonography
+
+<p align="center">
+    <img src="images/stegonography.png"/>
+</p>
+
+Stegonography is a Node.js web application that uses an API and a simple front end to simulate a real steganography application. It is built with two main routes, it's home( `/` ) and (`/login`), where the admin can log in to make changes to the page.
+
+## Index
+
+- [Definition](#what-is-security-misconfiguration)
+- [Setup](#setup)
+- [Attack narrative](#attack-narrative)
+- [Objectives](#secure-this-app-🔧)
+- [Solutions](#pr-solutions)
+- [Contributing](#contributing)
+
+## What is Security Misconfiguration?
+
+Security misconfiguration can happen at any level of an application stack, including the network services, platform, web server, application server, database, frameworks, custom code, and pre-installed virtual machines, containers, or storage. Automated scanners are useful for detecting misconfigurations, use of default accounts or configurations, unnecessary services, legacy options, etc.
+
+The main goal of this project is to discuss how **Security Misconfiguration** vulnerabilities can be exploited and to encourage developers to send secDevLabs Pull Requests  on how they would mitigate these flaws.
+
+## Setup
+
+To start this intentionally **insecure application**, you will need [Docker][Docker Install] and [Docker Compose][Docker Compose Install]. After forking [secDevLabs](https://github.com/globocom/secDevLabs), you must type the following commands to start:
+
+```sh
+cd secDevLabs/owasp-top10-2017-apps/a6/stegonography
+```
+
+```sh
+make install
+```
+
+Then simply visit [localhost:10006][App] ! 😆
+
+## Get to know the app 🦕
+
+To properly understand how this application works, you can follow these simple steps:
+
+- Try adding an hidden message to an image.
+- Try hiding a message with a password! 
+- Decrypt a friends message!
+
+## Attack narrative
+
+Now that you know the purpose of this app, what could possibly go wrong? The following section describes how an attacker could identify and eventually find sensitive information about the app or it's users. We encourage you to follow these steps and try to reproduce them on your own to better understand the attack vector! 😜
+
+### Vulnerability identification  👀
+
+#### Error stack traces are output to end users
+
+An attacker, when trying to enumerate available pages on the application, could come across a verbose error stack trace with potentially sensitive information that could compromise the app. An example of a verbose error stack trace is as shown by the image below:
+
+<p align="center">
+    <img src="images/stack_trace.png"/>
+</p>
+
+#### Default username and passwords are being used in a production environment 
```

Comment:
As the purpose of this app is to be run locally, "production environment" sounds weird to me. I do understand that you are trying to describe a fictitious scenario but what do you think of: "Default admin credentials"

Reply:
It does look a bit strange, thanks @rafaveira3 !